File: src/main/java/core/procedures/delayed/FutureCancel.java
package core.procedures.delayed;

import core.procedures.AFn;
import core.procedures.FnArgsBuilder;

import java.util.concurrent.Future;

public final class FutureCancel extends AFn {

  public FutureCancel() {
    super(new FnArgsBuilder().min(1).max(1).mandatory(new Class[]{Future.class}).build());
  }

  @Override
  public String getName() {
    return "future-cancel";
  }

  @Override
  public Boolean apply1(Object arg) {
    return ((Future)arg).cancel(true);
  }
}

File: src/main/java/core/procedures/delayed/FutureCancel.kt
package core.procedures.delayed

import core.procedures.AFn
import core.procedures.Arity.Exactly

import java.util.concurrent.Future

class FutureCancel : AFn<Future<*>?, Boolean>(name = "future-cancel", arity = Exactly(1),
                                              mandatoryArgsTypes = arrayOf(Future::class.java)) {

    override operator fun invoke(arg: Future<*>?) = arg!!.cancel(true)
}
